tests/Unit/CacheInvalidatorTest.php

```diff
@@ -184,8 +184,29 @@ public function testMethodException()
      */
     public function testProxyClientExceptionsAreLogged()
     {
-        $unreachableException = ProxyUnreachableException::proxyUnreachable('http://127.0.0.1', 'Couldn\'t connect to host');
-        $responseException    = ProxyResponseException::proxyResponse('http://127.0.0.1', 403, 'Forbidden');
+        $failedRequest = \Mockery::mock('\Psr\Http\Message\RequestInterface')
+            ->shouldReceive('getHeaderLine')->with('Host')->andReturn('127.0.0.1')
+            ->getMock();
+        $adapterException = \Mockery::mock(
+            '\Http\Adapter\Exception\HttpAdapterException',
+            ['Couldn\'t connect to host']
+        )
+            ->shouldReceive('getRequest')->andReturn($failedRequest)
+            ->getMock();
+        ;
+        $unreachableException = ProxyUnreachableException::proxyUnreachable($adapterException);
+        
+        $response = \Mockery::mock('\Psr\Http\Message\ResponseInterface')
+            ->shouldReceive('getStatusCode')->andReturn(403)
+            ->shouldReceive('getReasonPhrase')->andReturn('Forbidden')
+            ->getMock();
+        $adapterException = \Mockery::mock('\Http\Adapter\Exception\HttpAdapterException')
+            ->shouldReceive('getMessage')->andReturn('Forbidden')
+            ->shouldReceive('getRequest')->andReturn($failedRequest)
+            ->shouldReceive('getResponse')->andReturn($response)
+            ->getMock()
+        ;
+        $responseException = ProxyResponseException::proxyResponse($adapterException);
```

Comment:
Hey, great thoughts here. I'm a maintainer of php-vcr.  Thanks @dbu for pinging me about this discussion.

A MessageInterface to JSON/XML would be great. I was also thinking of moving php-vcr internally over to use PSR-7, but it makes more sense to create another library which can be used everywhere. 

Php-vcr has an btw [integration with phpunit](https://github.com/php-vcr/phpunit-testlistener-vcr) (but at the moment not phpspec).